trino/client.py

```diff
@@ -1128,14 +1128,39 @@ def map(self, values: List[Any]) -> Optional[List[Any]]:
         return [self.mapper.map(value) for value in values]
 
 
+class NamedRowTuple(tuple):
+    """Custom tuple class as namedtuple doesn't support missing or duplicate names"""
+    def __new__(cls, values, names: List[str], types: List[str]):
+        return super().__new__(cls, values)
+
+    def __init__(self, values, names: List[str], types: List[str]):
+        self._names = names
+        self.__annotations__ = dict()
```

Comment:
I think a comment here/commit message about how annotations are meant to be used (or why they are useful) would be useful.